crates/sui-types/src/temporary_store.rs

```diff
@@ -1123,6 +1125,32 @@ impl<'backing> ChildObjectResolver for TemporaryStore<'backing> {
                 .read_child_object(parent, child, child_version_upper_bound)
         }
     }
+
+    fn receive_object_at_version(
+        &self,
+        owner: &ObjectID,
+        receiving_object_id: &ObjectID,
+        receive_object_at_version: SequenceNumber,
+        epoch_id: EpochId,
+    ) -> SuiResult<Option<Object>> {
+        // You should never be able to try and receive an object after deleting it in the same
+        // transaction since `Receiving` doesn't have copy.
+        debug_assert!(self.deleted.get(receiving_object_id).is_none());
+        let obj_opt = self
+            .written
+            .get(receiving_object_id)
+            .map(|(obj, _kind)| obj);
+        if obj_opt.is_some() {
+            Ok(obj_opt.cloned())
```

Comment:
When is this case hit?

Reply:
<img src="https://media4.giphy.com/media/UPfaHtcHsU2td42eSb/giphy.gif"/>

This is actually impossible now that I'm looking at it since this would mean we're receiving an object that has been written which cannot happen. So just gonna delete this....